src/pages/text/[version].ts

```diff
@@ -0,0 +1,62 @@
+import type { APIRoute } from 'astro'
+import type { CollectionEntry, CollectionKey } from 'astro:content'
+import { getCollection } from 'astro:content'
+import { content } from '../../content'
+
+export const prerender = false
+
+type ContentEntry = CollectionEntry<
+  'core-docs' | 'quickstarts-docs' | 'react-component-docs'
+>
+
+export const GET: APIRoute = async ({ params }) => {
+  const { version } = params
+
+  if (!version) {
+    return new Response(
+      JSON.stringify({ error: 'Version parameter is required' }),
+      { status: 400, headers: { 'Content-Type': 'application/json' } },
```

Comment:
We should create a helper function to generate the HTTP headers.  Content-Type should also specify character encoding. Would set this to `application/json; charset=utf-8`